// multicapaRichiModule/index.js
const request = require('request');

constructor(libreria){

}

exports.printMsg = function() {
  console.log("Multicapa module found!");
}

exports.getUsers = function() {
	console.log('Haciendo peticion a /user/');
	request.get('http://www.richibuenaondaortizmulticapa.bid:3000/user/', function(e, users){
		console.log(users.body);
		return users;
	});
}

exports.getProducts = function() {
	console.log('Haciendo peticion a /productos/');
	request.get('http://www.richibuenaondaortizmulticapa.bid:3000/products/', function(e, products){
		console.log('Produtos en el nuevo cliente: ');
		console.log(products.body);
		return products;
	});
}

exports.postProducts = function(body) {
	console.log('Haciendo peticion a /productos/');
	console.log(body);
	let options = {
	    method: 'POST',
	    headers:{
	        //'Accept': '*/*',
	        //'Connection': 'keep-alive',
	        'Content-Type' : 'application/x-www-form-urlencoded'
	        //'User-Agent' : 'firefox/7.35.0'
	    },
	    url:'http://www.richibuenaondaortizmulticapa.bid:3000/productos/new'
    };
    options.form = body;
	request(options, function(e, response, body2){
		console.log(body2);
		return response;
	});
}
